shared/src/balancer/pool_cache.rs

```diff
@@ -13,10 +14,6 @@ use ethcontract::errors::MethodError;
 use ethcontract::{BlockId, Bytes, H160, H256, U256};
 use std::{collections::HashSet, sync::Arc};
 
-pub trait PoolCacheMetrics: Send + Sync {
```

Comment:
The idea with this extra indirection is we can have separate metrics for Balancer vs Uniswap pool fetching. It might be worth leaving this in.

Alternatively we can just make the metrics implement `CacheMetrics` directly and avoid the extra level of indirection with the `PoolCacheMetrics` trait.

Reply:
I don't really have a preference here. I would lean toward avoid extra direction, but I feel like this would be an unrelated change to this PR.

Reply:
Let's wait and see what others think, but IMO it is related to this PR as we are removing one of the traits here, so we should either remove both `PoolCacheMetrics` in order to remove one level of indirection, or keep the extra indirection and allow separate metrics to be used.

Reply:
I feel that it is useful to have separate metrics for separate pool fetching backends. Not very important, just slightly in favor.

Reply:
The reason I removed this is because there is a bit of a name conflict if we use `PoolCacheMetrics` because its easy to import the wrong one. I could also get around this by putting it back and calling it `WeightedPoolCacheMetrics`. I will do that for now.

Reply:
Cool. then I will keep them all separated, but rename this one to `WeightedPoolCacheMetrics` so that imports are not easily confused.